gpflow/logdensities.py

```diff
@@ -85,17 +83,15 @@ def multivariate_normal(x, mu, L):
     p[n] = log pdf of:
     x[n] ~ N(mu, LL^T) or x ~ N(mu[n], LL^T) or x[n] ~ N(mu[n], LL^T)
     """
-    mu_ndims = mu.shape.ndims
-    x_ndims = x.shape.ndims
-    if x_ndims is not None and x_ndims != 2:
-        raise ValueError('Shape of x must be 2D.')
-    if mu_ndims is not None and mu_ndims != 2:
-        raise ValueError('Shape of mu must be 2D.')
-
-    d = x - mu
-    alpha = tf.matrix_triangular_solve(L, d, lower=True)
-    num_dims = tf.cast(tf.shape(d)[0], L.dtype)
-    p = - 0.5 * tf.reduce_sum(tf.square(alpha), 0)
-    p -= 0.5 * num_dims * np.log(2 * np.pi)
-    p -= tf.reduce_sum(tf.log(tf.matrix_diag_part(L)))
-    return p
+    with tf.control_dependencies([
+            assert_tensor_ndim(x, 2, 'multivariate_normal requires the shape of x to be (N, D)'),
+            assert_tensor_ndim(mu, 2, 'multivariate_normal requires the shape of mu to be (N, D)'),
+            tf.assert_equal(tf.shape(L), [tf.shape(x)[0], tf.shape(x)[0]]),
```

Comment:
It's a bit confusing that you write in the two lines above that `x` and `mu` should be `(N, D)`, which implies that you check in this assert whether `L` is `(N, N)`, but in the docstrings it is stated that `L` should be `(D, D)`. Can you resolve this mismatch?

Reply:
The docstring says x and mu should be (D, N), so I think it's all correct as is?

Reply:
then change the `x to be (N, D)`?

Reply:
Sorry, you're absolutely right - it was confusing. I've changed it!